### DISCUSSION TITLE
DynamicLibrary.GetProcedure()

### DISCUSSION THREAD

**Discussion by D3Pheonix** [cleolibrary/CLEO-Redux#8] [Q&A] [answered]
How to use it ?
I have some scm code:
```
0AA4: 1@ = get_proc_address "GetLocalTime" library 0@ // IF and SET 
0AC8: 2@ = allocate_memory_size 32 
0AA5: call 1@ num_params 1 pop 0 2@ 
```
I want to translate it into JS, like this:
```
var localTimeAddress = libKernel.getProcedure("GetLocalTime");
var mem = Memory.Allocate(32);
Memory.CallFunction(localTimeAddress, 1, 0, mem); // ?? Crash
```

**Comment by x87** [comment, marked as answer]
Thanks :)

`DynamicLibrary.GetProcedure` has an issue, as CLEO 4 has unconventionally put `self` as the second argument in 0AA4 (https://library.sannybuilder.com/#/sa/CLEO/0AA4), so CLEO Redux may call this instruction with incorrect parameter order. I will take a look.

Meanwhile, you can use the opcode directly using `op` command: `var localTimeAddress = op(0x0AA4, "GetLocalTime", libKernel);`

**Reply by D3Pheonix** [reply]
And what about ```Memory.CallFunction```, i think, it also has an issue )

**Reply by x87** [reply]
what happens when you call Memory.CallFunction?

**Reply by D3Pheonix** [reply]
I'll check it today. 
P.S What about gta trilogy remastered? I mean cleo / redux library for this series. Have you any plans on them?

**Reply by x87** [reply]
![Maybe  - Imgur](https://user-images.githubusercontent.com/5698288/141901405-c54254f0-d55b-4858-befe-1e01d2da89dd.gif)

**Comment by x87** [comment]
>DynamicLibrary.GetProcedure has an issue

fixed in https://github.com/cleolibrary/CLEO-Redux/releases/tag/0.7.5
